BurnOutSharp/PackerType/MicrosoftCABSFX.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using BurnOutSharp.Matching;
+
+namespace BurnOutSharp.PackerType
+{
+    // TODO: Add extraction, which should be possible with LibMSPackN, but it refuses to extract due to SFX files lacking the typical CAB identifiers.
+    public class MicrosoftCABSFX : IContentCheck
+    {
+        /// <inheritdoc/>
+        public bool ShouldScan(byte[] magic) => true;
+
+        /// <inheritdoc/>
+        public string CheckContents(string file, byte[] fileContent, bool includePosition = false)
+        {
+
+            var fileDetails = Utilities.GetFileVersionInfo(file);
+
+            string name = fileDetails?.InternalName.Trim();
+            if (!string.IsNullOrWhiteSpace(name) && name.Equals("Wextract", StringComparison.OrdinalIgnoreCase))
+                return $"Microsoft CAB SFX v{Utilities.GetFileVersion(file)}";
+
+            name = fileDetails?.OriginalFilename.Trim();
+            if (!string.IsNullOrWhiteSpace(name) && name.Equals("WEXTRACT.EXE", StringComparison.OrdinalIgnoreCase))
+                return $"Microsoft CAB SFX v{Utilities.GetFileVersion(file)}";
+
+            var matchers = new List<ContentMatchSet>
+            {
+                // wextract_cleanup
+                new ContentMatchSet(new byte?[]
+                {
+                    0x77, 0x65, 0x78, 0x74, 0x72, 0x61, 0x63, 0x74, 
+                    0x5F, 0x63, 0x6C, 0x65, 0x61, 0x6E, 0x75, 0x70
+                }, GetVersion, "Microsoft CAB SFX"),
+
+                // MSCFu
+                new ContentMatchSet(new byte?[] { 0x4D, 0x53, 0x43, 0x46, 0x75 }, GetVersion, "Microsoft CAB SFX (Generic?)"),
```

Comment:
Do you need to have `(Generic?)` there? Seems... odd. Because you'll end up with strings that look like `Microsoft CAB SFC (Generic?) (1.0.2123)` in the output